### PR TITLE
Add unit tests for OpenmrsUtil.isStringInArray

### PR DESCRIPTION
**Pull request by varshithreddybokka4444-netizen**
## Description of what I changed

Added unit tests for `OpenmrsUtil.isStringInArray` to cover standard and edge cases, including:
- when the string exists in the array
- when the string does not exist
- null string input
- null array input
- arrays containing null values

These tests document and lock the expected behavior of the utility method.

## Issue I worked on

N/A – test-only internal improvement (no JIRA issue required)

## Checklist: I completed these to help reviewers :)

- [ ] My IDE is configured to follow the code style of this project.

- [x] I have **added tests** to cover my changes.

- [ ] I ran `mvn clean package` right before creating this pull request.

  _Note: The modified tests were run successfully via IntelliJ. Full test suite will be executed by CI._

- [x] All new and existing **tests passed**.

- [ ] My pull request is **based on the latest changes** of the master branch.

### CODE REVIEW

**Comment by varshithreddybokka4444-netizen** [comment]
Hi, this is my first contribution to OpenMRS.
Please let me know if any changes are required.
Thanks!

File: api/src/test/java/org/openmrs/util/OpenmrsUtilTest.java [hunk 1]
```diff
@@ -688,7 +688,51 @@ public void shortenedStackTrace_shouldReturnNullIfStackTraceIsNull() {
...
+	public void isStringInArray_shouldReturnFalseIfStringDoesNotExistInArray() {
+		String[] arr = {"a", "b", "c"};
+		assertFalse(OpenmrsUtil.isStringInArray("d", arr));
+	}
+
+	/**
+	 * @see OpenmrsUtil#isStringInArray(String, String[])
+	 */
+	@Test
+	public void isStringInArray_shouldReturnFalseIfStringIsNull() {
```
Comment (ibacher) [inline api/src/test/java/org/openmrs/util/OpenmrsUtilTest.java:709]: Two additional cases to consider:

```java
String[] arr = {"a", null, "b"};
OpenmrsUtil.isStringInArray(null, arr);
```

```java
OpenmrsUtil.isStringInArray(null, null);
```

Comment (varshithreddybokka4444-netizen) [reply api/src/test/java/org/openmrs/util/OpenmrsUtilTest.java:709]: Thanks for the suggestion! I’ve added tests to cover both of these cases.
Please let me know if any further adjustments are needed.

**Review by arvindtelharkar2024 — APPROVED** [review]

File: api/src/test/java/org/openmrs/util/OpenmrsUtilTest.java [hunk 2]
```diff
@@ -688,7 +688,67 @@ public void shortenedStackTrace_shouldReturnNullIfStackTraceIsNull() {
 	public void nullSafeEqualsIgnoreCase_shouldBeCaseInsensitive() {
 		assertTrue(OpenmrsUtil.nullSafeEqualsIgnoreCase("equal", "Equal"));
 	}
```
Comment (wikumChamith) [inline api/src/test/java/org/openmrs/util/OpenmrsUtilTest.java:685]: Add a blank line before a new test starts.

File: api/src/test/java/org/openmrs/util/OpenmrsUtilTest.java [hunk 3]
```diff
@@ -688,7 +688,67 @@ public void shortenedStackTrace_shouldReturnNullIfStackTraceIsNull() {
...
+	}
+
+	/**
+	 * @see OpenmrsUtil#isStringInArray(String, String[])
+	 */
+	@Test
+	public void isStringInArray_shouldHandleNullValuesInsideArray() {
+		String[] arr = {"a", null, "b"};
+		assertTrue(OpenmrsUtil.isStringInArray("a", arr));
+		assertFalse(OpenmrsUtil.isStringInArray("c", arr));
```
Comment (wikumChamith) [inline api/src/test/java/org/openmrs/util/OpenmrsUtilTest.java:729]: Do we need this here? Feels like this test is doing two things at once.

File: api/src/test/java/org/openmrs/util/OpenmrsUtilTest.java (hunk 2 above)
Comment (varshithreddybokka4444-netizen) [reply api/src/test/java/org/openmrs/util/OpenmrsUtilTest.java:685]: i have now added the blank line, thank you for correcion.

File: api/src/test/java/org/openmrs/util/OpenmrsUtilTest.java (hunk 3 above)
Comment (varshithreddybokka4444-netizen) [reply api/src/test/java/org/openmrs/util/OpenmrsUtilTest.java:729]: Thank you for correcting, i removed the line, please check now. Please sugest any more changes if needed.

**Comment by wikumChamith** [comment]
@varshithreddybokka4444-netizen can you resolve the merge conflicts?

**Comment by varshithreddybokka4444-netizen** [comment]
will resolve the merge conflicts as fast as possible

**Comment by varshithreddybokka4444-netizen** [comment]
@wikumChamith i resolved conflicts, please check and suggest the changes if any

**Comment by ibacher** [comment]
@varshithreddybokka4444-netizen It doesn't look like a couple of the updates you made landed. Can you ensure you've properly pushed to your branch?
